src/main/java/io/vertx/httpproxy/interceptors/impl/HeadersInterceptorImpl.java

```diff
@@ -11,10 +11,50 @@
 
 package io.vertx.httpproxy.interceptors.impl;
 
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.httpproxy.ProxyContext;
 import io.vertx.httpproxy.ProxyInterceptor;
+import io.vertx.httpproxy.ProxyRequest;
+import io.vertx.httpproxy.ProxyResponse;
 
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * The general interceptor for headers.
+ */
 public class HeadersInterceptorImpl implements ProxyInterceptor {
+  private final Handler<MultiMap> changeRequestHeaders;
+  private final Handler<MultiMap> changeResponseHeaders;
+  private static final Handler<MultiMap> NO_OP = mmap -> {};
```

Comment:
For readability, it's better to separate constants from fields

```suggestion

  private static final Handler<MultiMap> NO_OP = mmap -> {};

  private final Handler<MultiMap> changeRequestHeaders;
  private final Handler<MultiMap> changeResponseHeaders;
```